### ISSUE TITLE
NuGet does not deal with blocking version conflicts from existing installed packages

### ISSUE THREAD

**Issue by ferventcoder** [chocolatey/choco#116]
1. Conflicting dependencies - When installing a package that has dependencies on a newer version of a package than are allowed by an existing package with that dependency - it just allows the install/upgrade without any warning.

**Comment by ferventcoder** [comment]
:panda_face: :cry:

**Comment by ferventcoder** [comment]
I believe this is because NuGet doesn't seem to take into account all of the existing packages in the local repository.

**Comment by gregsdennis** [comment]
I, too, just experienced this.  Our application needs this to be handled, can the priority on this be bumped up?

In the meantime, we'll have to write custom logic to facilitate this.

**Comment by ferventcoder** [comment]
It was a lower priority at one point - once we realized it has other issues, we bumped the priority of this (but not on the ticket). Follow #508 where I will be hopefully taking on a newer dependency of NuGet that does better at this. Otherwise we will start working into the dependency resolver and walker and making it know about every existing package as well because this is just... just not good.

**Comment by ferventcoder** [comment]
#507 is a hoot as well :-1:

**Comment by ferventcoder** [comment]
This requires .NET 4.5, so will need to be implemented with 0.9.11.

**Comment by jberezanski** [comment]
Is this still on the radar? I just ran into it while testing new .NET Core packages.

Given this dependency graph:
`dotnetcore-3.0-windowshosting` 3.0.3 -> `<dependency id="dotnet-aspnetcoremodule-v2" version="[13.0.20023,13.0.20024)" />`
`dotnetcore-3.1-windowshosting` 3.1.9 -> `<dependency id="dotnet-aspnetcoremodule-v2" version="[13.1.20268,13.1.20269)" />`

this sequence:
```
cinst -y dotnetcore-3.0-windowshosting
cinst -y dotnetcore-3.1-windowshosting
```
should fail on the second choco invocation, but instead proceeds to upgrade dotnet-aspnetcoremodule-v2 to 13.1.20268, violating the dependency constraint of dotnetcore-3.0-windowshosting.

**Comment by jberezanski** [comment]
From https://github.com/NuGet/Home/issues/1870#issuecomment-169192305 it seems it could be fixed even with NuGet v2, without spending the major effort of upgrading to NuGet v3+.

**Comment by TheCakeIsNaOH** [comment]
This appears to have been fixed as a part of #2740, and the below test was enabled to cover it:
https://github.com/chocolatey/choco/blob/c0e2f848df3ba61b3a74504e8650020b588a6b16/src/chocolatey.tests.integration/scenarios/InstallScenarios.cs#L3136-L3249
